src/Message.ts

```diff
@@ -254,3 +242,126 @@ export class MessageV2 extends MessageBase implements proto.MessageV2 {
 }
 
 export type Message = MessageV1 | MessageV2
+
+export interface DecodedMessage {
+  id: string
+  messageVersion: 'v1' | 'v2'
+  senderAddress: string
+  conversation: Conversation
+  contentType: ContentTypeId
+  sent: Date
+  contentTopic: string
+  recipientAddress?: string
```

Comment:
I have doubts about recipientAddress too, but it may not be easy to get rid of it.

Reply:
It definitely has more of a chance of breaking downstream apps, but we do all our checks based on recipient address before we convert to a `DecodedMessage` (I think, possibly not true in tests) so it's safe to do on our end.